packages/web-components/src/components/back-to-top/back-to-top.ts

```diff
@@ -0,0 +1,158 @@
+ *
+ * This source code is licensed under the Apache-2.0 license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+import { html, property, customElement, LitElement } from 'lit-element';
+import ddsSettings from '@carbon/ibmdotcom-utilities/es/utilities/settings/settings.js';
+import HostListener from 'carbon-web-components/es/globals/decorators/host-listener';
+import HostListenerMixin from 'carbon-web-components/es/globals/mixins/host-listener';
+import settings from 'carbon-components/es/globals/js/settings.js';
+import throttle from 'lodash-es/throttle.js';
+import UpToTop20 from 'carbon-web-components/es/icons/up-to-top/20.js';
+import styles from './back-to-top.scss';
+
+const { prefix } = settings;
+const { stablePrefix: ddsPrefix } = ddsSettings;
+interface Cancelable {
+  cancel(): void;
+}
+
+/**
+ * Back to top
+ *
+ * @element dds-back-to-top
+ */
+@customElement(`${ddsPrefix}-back-to-top`)
+class DDSBackToTop extends HostListenerMixin(LitElement) {
+  /**
+   * The observer for the resize of the viewport.
+   */
+  private _observerResizeRoot: any | null = null; // TODO: Wait for `.d.ts` update to support `ResizeObserver`
+
+  /**
+   * The document height
+   */
+  private _bodyHeight!: number;
+
+  /**
+   * The viewport height
+   */
+  private _windowHeight!: number;
+
+  /**
+   * The handler for throttled scrolling
+   */
+  private _throttleScroll: (((event: Event) => void) & Cancelable) | null = null;
+
+  /**
+   * Button click scrolls to top
+   */
+  // eslint-disable-next-line class-methods-use-this
+  private _handleOnClick() {
+    this.ownerDocument!.defaultView!.scrollTo({ top: 0, behavior: 'smooth' });
+  }
+
+  /**
+   * Button visible when one full page length is scrolled if _showBackToTop is true
+   */
+  private _handleOnScroll() {
+    if (this._showBackToTop()) {
+      if (document.body.scrollTop > this._windowHeight || document.documentElement.scrollTop > this._windowHeight) {
+        this.hidden = false;
+      } else {
+        this.hidden = true;
+      }
+    }
+  }
+
+  /**
+   * Cleans-up and creats the resize observer for the scrolling container.
+   *
+   * @param [options] The options.
+   * @param [options.create] `true` to create the new resize observer.
+   */
+  private _cleanAndCreateObserverResize({ create }: { create?: boolean } = {}) {
+    if (this._observerResizeRoot) {
+      this._observerResizeRoot.disconnect();
+      this._observerResizeRoot = null;
+    }
+    if (create) {
+      // TODO: Wait for `.d.ts` update to support `ResizeObserver`
+      // @ts-ignore
+      this._observerResizeRoot = new ResizeObserver(this._observeResizeRoot);
+      this._observerResizeRoot.observe(this.ownerDocument!.documentElement);
+    }
+    // }
+  }
+
+  private _observeResizeRoot = () => {
+    this._bodyHeight = this.ownerDocument!.documentElement.scrollHeight;
+    this._windowHeight = this.ownerDocument!.documentElement.clientHeight;
+  };
```

Comment:
And we need another `ResizeObserver` that looks at `<body>`, and set `._bodyHeight` in the new observer.

```suggestion
  private _observeResizeRoot = entries => {
    this._windowHeight = entries[entries.length - 1].contentRect.height;
  };
```